src/basic/socket-util.h

```diff
@@ -105,11 +105,11 @@ const char* socket_address_get_path(const SocketAddress *a);
 bool socket_ipv6_is_supported(void);
 bool socket_ipv6_is_enabled(void);
 
-int sockaddr_port(const struct sockaddr *_sa, unsigned *port);
+int sockaddr_port(const union sockaddr_union *sa, unsigned *port);
```

Comment:
i am really not convinced this is a good change. The way I see it our `union sockaddr_union` is mostly just a construct we use to allocate enough memory for any type of sockaddr we could possibly care about. It's thus mostly used for *allocation*, but not for *consumption* of these structures, because libc after all just returns `struct sockaddr` to us.

In modern C it's generally not OK to have a pointer to a larger type point to memory allocated for a shorter type, but the reverse is OK. i.e. compilers and static analyzers have been reserving the right to assume that if something is a large type (i.e. union sockaddr_union), then every single bit of it is in valid memory. But this is not necessarily the case if we ever cast a struct sockaddr into a union sockaddr_union. The reverse cast is OK however.

Hence, I think this change is pretty much not OK.

Reply:
(i am fully aware btw, that some of our functions already do such a cast. I am somewhat comfortable with that hwoever, since it's generally encapsulated in some inner function, and not exposed in apis)

Reply:
Alright, I addressed this issue. Thank you!